src/utils/proxy.ts

```diff
@@ -433,9 +437,13 @@ const serveRedirect = async function ({
       match.force ||
       (!staticFile && ((!options.framework && destStaticFile) || isInternal(destURL) || matchingFunction))
     ) {
+      // 3xx redirects parsed above, here are 2xx meaning just override the url of proxying page and use the status
+      // which comes from that url
       req.url = destStaticFile ? destStaticFile + dest.search : destURL
       const { status } = match
-      statusValue = status
+      if (match.force || status !== 200) {
```

Comment:
override status code only if forcing or status is not 200, when 200 is just shadowing the page by the url, so it's status should be propagated (e.g. 404 because it's not there or idk 201 because server decided to return 201 by that url)